js/ui/tape.ts

```diff
@@ -36,7 +36,7 @@ export default class Tape {
         fileReader.onload = (ev: ProgressEvent) => {
             const target: FileReader = ev.target as FileReader;
             const result: ArrayBuffer = target.result as ArrayBuffer;
-            context.decodeAudioData(result, (buffer) => {
+            context.decodeAudioData(result).then((buffer) => {
```

Comment:
Maybe apply the same transformation you did above instead:
```
context.context.decodeAudioData(result).then((buffer) => {...}).catch(console.error);
```
(Or `window.alert` if you prefer.) Or just make it `async`:
```typescript
fileReader.onload = async (ev: ProgressEvent) => {
    ...
    try {
        const buffer = await context.decodeAudioData(result);
        ...
    } catch (e) {
        window.alert(e);
    }
```